src/main/java/org/checkerframework/checker/androidresource/AndroidResourceAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,402 @@
+    /**
+     * The @FractionContainer annotation.
+     */
+    protected final AnnotationMirror FRACTION_CONTAINER =
+            AnnotationBuilder.fromClass(elements, FractionContainer.class);
+    /**
+     * The @IdContainer annotation.
+     */
+    protected final AnnotationMirror ID_CONTAINER =
+            AnnotationBuilder.fromClass(elements, IdContainer.class);
+    /**
+     * The @IntegerContainer annotation.
+     */
+    protected final AnnotationMirror INTEGER_CONTAINER =
+            AnnotationBuilder.fromClass(elements, IntegerContainer.class);
+    /**
+     * The @InterpolatorContainer annotation.
+     */
+    protected final AnnotationMirror INTERPOLATOR_CONTAINER =
+            AnnotationBuilder.fromClass(elements, InterpolatorContainer.class);
+    /**
+     * The @LayoutContainer annotation.
+     */
+    protected final AnnotationMirror LAYOUT_CONTAINER =
+            AnnotationBuilder.fromClass(elements, LayoutContainer.class);
+    /**
+     * The @MenuContainer annotation.
+     */
+    protected final AnnotationMirror MENU_CONTAINER =
+            AnnotationBuilder.fromClass(elements, MenuContainer.class);
+    /**
+     * The @PluralsContainer annotation.
+     */
+    protected final AnnotationMirror PLURALS_CONTAINER =
+            AnnotationBuilder.fromClass(elements, PluralsContainer.class);
+    /**
+     * The @RawContainer annotation.
+     */
+    protected final AnnotationMirror RAW_CONTAINER =
+            AnnotationBuilder.fromClass(elements, RawContainer.class);
+    /**
+     * The @StringContainer annotation.
+     */
+    protected final AnnotationMirror STRING_CONTAINER =
+            AnnotationBuilder.fromClass(elements, StringContainer.class);
+    /**
+     * The @StyleableContainer annotation.
+     */
+    protected final AnnotationMirror STYLEABLE_CONTAINER =
+            AnnotationBuilder.fromClass(elements, StyleableContainer.class);
+    /**
+     * The @XmlContainer annotation.
+     */
+    protected final AnnotationMirror XML_CONTAINER =
+            AnnotationBuilder.fromClass(elements, XmlContainer.class);
+
+    public AndroidResourceAnnotatedTypeFactory(BaseTypeChecker checker) {
+        super(checker);
+        postInit();
+    }
+
+    @Override
+    protected TreeAnnotator createTreeAnnotator() {
+        return new ListTreeAnnotator(
+                new AndroidResourceTreeAnnotator(this), super.createTreeAnnotator());
+    }
+
+    private class AndroidResourceTreeAnnotator extends TreeAnnotator {
+
+        public AndroidResourceTreeAnnotator(AnnotatedTypeFactory annotatedTypeFactory) {
+            super(annotatedTypeFactory);
+        }
+
+        /*
+         * Method for a tree node for a member access expression. For example:
+         *       expression . identifier
+         */
+        @Override
+        public Void visitMemberSelect(MemberSelectTree node, AnnotatedTypeMirror annotatedTypeMirror) {
+
+            String expression = node.getExpression().toString();

```

Comment:
Why not move these two variables into `assignContainerAnnotations` and make it a more local decision how it is implemented?

Reply:
Fixed.